docs/administration-guide/named-operations.md

```diff
@@ -0,0 +1,906 @@
+
+    === "Java"
+
+        ``` java
+        final NamedOperation<EntityId, CloseableIterable<EntityId>> operation =
+        new NamedOperation.Builder<EntityId, CloseableIterable<EntityId>>()
+                .name("2-hop")
+                .input(new EntitySeed(1))
+                .build();
+        ```
+
+    === "JSON"
+
+        ``` json
+        {
+            "class" : "NamedOperation",
+            "input" : [ {
+                "class" : "EntitySeed",
+                "class" : "EntitySeed",
+                "vertex" : 1
+            } ],
+            "operationName" : "2-hop"
+        }
+        ```
+
+    === "Python"
+
+        ``` python
+        g.NamedOperation( 
+            input=[ 
+                g.EntitySeed( 
+                    vertex=1 
+                ) 
+            ], 
+            operation_name="2-hop" 
+        )
+        ```
+
+
+This produces the following result:
+```java
+EntitySeed[vertex=4]
+EntitySeed[vertex=3]
+EntitySeed[vertex=5]
+```
+
+Named Operations are able to take parameters which allow the OperationChain that is being executed to be configured.
+Parameters could be as simple as specifying `resultLimit` on a Limit Operation, they could specify a custom view to use in an Operation or even the input to an Operation.
+
+When adding a NamedOperation with parameters to an OperationChain it must be specified as a JSON string, with the parameter names enclosed in `${` and `}`. 
+For each parameter, a `ParameterDetail` object must be created which gives a description, a class type and an optional default for that parameter.
+As the default is optional you can alternatively indicate that the parameter must be provided and that there is no default.
+
+The following code adds a NamedOperation with a parameter that allows the result limit for the OperationChain to be set:
+
+!!! example "Adding a NamedOperation with parameters"
+
+    === "Java"
+
+        ``` java
+        final String opChainString = new JSONObject()
+            .put("operations", new JSONArray()
+                .put(new JSONObject()
+                    .put("class", "GetAdjacentIds")
+                    .put("includeIncomingOutGoing", "OUTGOING"))
+                .put(new JSONObject()
+                    .put("class", "GetAdjacentIds")
+                    .put("includeIncomingOutGoing", "OUTGOING"))
+                .put(new JSONObject()
+                    .put("class", "Limit")
+                    .put("resultLimit", "${param1}")))
+            .toString();
+
+        ParameterDetail param = new ParameterDetail.Builder()
+                .defaultValue(1L)
+                .description("Limit param")
+                .valueClass(Long.class)
+                .build();
+        Map<String, ParameterDetail> paramMap = Maps.newHashMap();
+        paramMap.put("param1", param);
+
+        final AddNamedOperation operation = new AddNamedOperation.Builder()
+                .operationChain(opChainString)
```

Comment:
No it can be done using an OperationChain object if you wanted :) https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/named/operation/AddNamedOperation.java#L275-L283